source/extensions/quic_listeners/quiche/platform/quic_mem_slice_impl.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include <iostream>
+#include <memory>
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+
+#include "quiche/quic/core/quic_buffer_allocator.h"
+
+namespace quic {
+
+class QuicMemSliceImpl {
+public:
+  // Constructs an empty QuicMemSliceImpl.
+  QuicMemSliceImpl() = default;
+
+  // Constructs a QuicMemSliceImpl by let |allocator| allocate a data buffer of
+  // |length|.
+  QuicMemSliceImpl(QuicBufferAllocator* allocator, size_t length)
+      : length_(length), slice_(new Envoy::Buffer::OwnedImpl()) {
+    auto fragment = new Envoy::Buffer::BufferFragmentImpl(
+        allocator->New(length), length,
+        [allocator](const void* data, size_t, const Envoy::Buffer::BufferFragmentImpl* fragment) {
+          allocator->Delete(const_cast<char*>(static_cast<const char*>(data)));
+          delete fragment;
```

Comment:
i'm surprised this 'delete' compiles since it's passed as a const pointer.

Reply:
deleter can still be called for const object so that it can be destructed.